File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit=Integer.MIN_VALUE;
        int minPrice=Integer.MAX_VALUE;
        for(int sell: prices){
            minPrice=Math.min(minPrice,sell);
            maxProfit=Math.max(maxProfit,sell-minPrice);
        }
        return maxProfit;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        lt=0
        rt=len(heights)-1
        maxlen=-float("inf")
        maxarea=-float("inf")
        while rt>lt:
            if(heights[rt]>=heights[lt]):
                maxlen=rt-lt
                maxarea=max(maxarea,maxlen*min(heights[lt],heights[rt]))
                lt+=1
            else:
                maxlen=rt-lt
                maxarea=max(maxarea,maxlen*min(heights[lt],heights[rt]))
                rt-=1
        return maxarea